webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -498,6 +517,33 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
     [updateSettings],
   );
 
+  const handleHeatmapChange = useCallback(
+    (selection: string[]) => {
+      updateSettings({ heatmapSkipped: selection });
+    },
+    [updateSettings],
+  );
+
+  const handleToggleHeatmap = useCallback(
+    (heatmapOn: boolean) => {
+      updateSettings({ heatmapOn: !heatmapOn });
+    },
+    [updateSettings],
+  );
+
+  const heatmapBtnVisible = useMemo(() => {
+    return !!Loadable.getOrElse([], projectColumns).find(
+      (column) =>
+        settings.columns
+          .slice(0, settings.compare ? settings.pinnedColumnsCount : undefined)
+          .includes(column.column) &&
+        (column.column === 'searcherMetricsVal' ||
+          (column.type === V1ColumnType.NUMBER &&
+            (column.location === V1LocationType.VALIDATIONS ||
+              column.location === V1LocationType.TRAINING))),
+    );
```

Comment:
nit: instead of converting the found element a boolean, you could use `.some`. Also, consider moving the settings column calculation out of the loop
```suggestion
    const visibleColumns = settings.columns
          .slice(0, settings.compare ? settings.pinnedColumnsCount : undefined)
    return Loadable.getOrElse([], projectColumns).some(
      (column) =>
        visibleColumns.includes(column.column) &&
        (column.column === 'searcherMetricsVal' ||
          (column.type === V1ColumnType.NUMBER &&
            (column.location === V1LocationType.VALIDATIONS ||
              column.location === V1LocationType.TRAINING))),
    );
```
This is out of scope for this pr, but it seems like we check the type of a column from the settings column a few times.  we should consider having a memo of the project columns mapped by id so we don't have to search the array of project columns every time.